cmd/juju/gui/gui.go

```diff
@@ -151,6 +154,20 @@ func (c *guiCommand) Run(ctx *cmd.Context) error {
 	return nil
 }
 
+// guiAddr returns an address where the GUI is available.
+func guiAddr(ctx *cmd.Context, conn api.Connection) string {
```

Comment:
What's changed that this is now needed?

Reply:
Now the dns name is used if available, rather than the IP address.